NewsDistribution/NewsServer.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net;
+using System.Net.Sockets;
+
+namespace NewsDistribution;
+
+public class NetworkClient
+{
+    public readonly Socket Socket;
+    public readonly string Name;
+
+    public NetworkClient(Socket socket, string name)
+    {
+        Socket = socket;
+        Name = name;
+    }
+}
+
+public class NewsServer
+{
+    private readonly Dictionary<string, NetworkClient> _clients = new();
+    private Socket? _listener;
+    private Thread? _acceptThread;
+    private CancellationTokenSource? _shouldAcceptClients;
+
+    public Dictionary<string, NetworkClient> Clients
+    {
+        get => _clients;
+    }
+
+    public bool Start(ushort port)
+    {
+        if (_listener != null)
+            return false;
+
+        try
+        {
+            _listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            _listener.Bind(new IPEndPoint(IPAddress.Any, port));
+            _listener.Listen();
+
+            _shouldAcceptClients = new();
```

Comment:
Initializing fields out of context must explicitly specify the type of initialization.
Replace with ```_shouldAcceptClients = new CancellationTokenSource();```